src/lib.rs

```diff
@@ -742,12 +727,20 @@ fn parse(chars: &mut CharIter) -> Result<Requirement, Pep508Error> {
 
 /// A library for [dependency specifiers](https://packaging.python.org/en/latest/specifications/dependency-specifiers/)
 /// as originally specified in [PEP 508](https://peps.python.org/pep-0508/)
+///
+/// This has `Version` and `VersionSpecifier` included. That is because
+/// `pep440_rs.Version("1.2.3") != pep508_rs.Requirement("numpy==1.2.3").version_or_url` as the
+/// `Version`s come from two different binaries and can therefore never be equal.
 #[cfg(feature = "pyo3")]
 #[pymodule]
-pub fn pep508_rs(py: Python<'_>, m: &PyModule) -> PyResult<()> {
+#[pyo3(name = "pep508_rs")]
+pub fn python_module(py: Python<'_>, m: &PyModule) -> PyResult<()> {
     #[cfg(feature = "pyo3")]
     pyo3_log::init();
 
+    m.add_class::<Version>()?;
+    m.add_class::<VersionSpecifier>()?;
```

Comment:
I don't think I fully understand the fix or why it is working. 

I understand that equality isn't given between `Version` and `VersionSpecifier`s coming from `pep440` and `pep508` because python doesn't understand that these are the same types (because they come from two different binaries). 

Why is `add_class` fixing this? My first thought was that it now creates a new `Version` and `Versionspecifier` name specific to `pep508`.

Reply:
yes, that's indeed the workaround i was going for. previously,

```
from pep440_rs import Version
from pep508_rs import Requirement

assert Requirement("numpy==1.2.3").version == Version("numpy==1.2.3")
```

would fail, and there way no way to construct an equivalent `Version` for pep508_rs. With this added, we get:

```
from pep508_rs import Version, Requirement

assert Requirement("numpy==1.2.3").version == Version("numpy==1.2.3")
```

which passes. admittedly, this means the `pep508_rs` python package is now effectively PEP 508 + PEP 440 in one package, but unfortunately i don't really have a better idea that doesn't involve either stringifying/reparsing version comparisons as they were before or writing c headers (i tried adding a `pep440` submodule before but that was worse than just adding the two types)

Reply:
I see. Does that mean that (sorry for my basic python :D)

```python
from pep508_rs import Pep508Version
from pep440_rs import Version

assert Version("numpy==1.2.3") == Pep508Version("numpy==1.2.3")
```

throws?

Reply:
yep! that's the exact case i ran into in my unit tests